### PR TITLE
Implement exit proposal in validator client

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

This PR implements `ProposeExit` in the validator client and includes tests for the implemented functionality.

**Which issues(s) does this PR fix?**

Part of #6882 

**Other notes for review**

- The client is not connected to the CLI command yet. This will come later.
- There are some duplications and redundancies in the submitted files. I will clean them up in a separate PR after this one is merged.

### CODE REVIEW

File: validator/client/metrics.go [hunk 1]
```diff
@@ -70,6 +70,28 @@ var (
 			"pubkey",
 		},
 	)
+	// ValidatorProposeExitSuccessVec used to count successful voluntary exit proposals.
+	ValidatorProposeExitSuccessVec = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "validator",
+			Name:      "successful_exit_proposals",
```
Comment (terencechain) [inline validator/client/metrics.go:77]:
```suggestion
			Name:      "successful_exit_proposals_total",
```

Let's follow the standard: https://prometheus.io/docs/practices/naming/

File: validator/client/metrics.go [hunk 2]
```diff
@@ -70,6 +70,28 @@ var (
...
+		},
+		[]string{
+			// validator pubkey
```
Comment (terencechain) [inline validator/client/metrics.go:80]:
```suggestion
```

Don't think we need a comment for this

File: validator/client/propose.go [hunk 3]
```diff
@@ -203,3 +234,42 @@ func (v *validator) signBlock(ctx context.Context, pubKey [48]byte, epoch uint64
...
+
+// Sign voluntary exit with proposer domain and private key.
+func (v *validator) signVoluntaryExit(ctx context.Context, pubKey [48]byte, exit *ethpb.VoluntaryExit) ([]byte, error) {
+	domain, err := v.domainData(ctx, exit.Epoch, params.BeaconConfig().DomainVoluntaryExit[:])
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get domain data")
+	}
+	var sig bls.Signature
+
+	exitRoot, err := helpers.ComputeSigningRoot(exit, domain.SignatureDomain)
```
Comment (rauljordan) [inline validator/client/propose.go:246]: can domain ever be nil?

Comment (rkapka) [reply validator/client/propose.go:246]: Wouldn't hurt to check this since it's coming from an RPC call.

**Comment by rkapka** [comment]
I merged this PR's branch with the branch from #7122, so latest changes include additionally `exit.go` and `exit_test.go`.

**Pull request merged**